hwci/boards/tockloader_board.py

```diff
@@ -19,28 +19,38 @@ def __init__(self):
             os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
         )
 
-    def flash_app(self, app):
-        logging.info(f"Flashing app: {app}")
+    def flash_app(self, app_path):
+        app_name = os.path.basename(app_path)
+        logging.info(f"Flashing app: {app_name}")
         libtock_c_dir = os.path.join(self.base_dir, "libtock-c")
         if not os.path.exists(libtock_c_dir):
             logging.error(f"libtock-c directory {libtock_c_dir} not found")
             raise FileNotFoundError(f"libtock-c directory {libtock_c_dir} not found")
 
-        app_dir = os.path.join(libtock_c_dir, "examples", app)
+        app_dir = os.path.join(libtock_c_dir, "examples", app_path)
         if not os.path.exists(app_dir):
             logging.error(f"App directory {app_dir} not found")
             raise FileNotFoundError(f"App directory {app_dir} not found")
 
         # Build the app using absolute paths
-        logging.info(f"Building app: {app}")
-        subprocess.run(["make", f"TOCK_TARGETS={self.arch}"], cwd=app_dir, check=True)
+        logging.info(f"Building app: {app_name}")
+        if app_name != "lua-hello":
+            subprocess.run(
+                ["make", f"TOCK_TARGETS={self.arch}"], cwd=app_dir, check=True
+            )
+        else:
+            # if the app is lua-hello, we need to build the libtock-c submodule first so we need to change directory
+            # into the libtock-c directory so it knows we are in a git repostiory
+            self.change_directory(libtock_c_dir)
+            subprocess.run(
+                ["make", f"TOCK_TARGETS={self.arch}"], cwd=app_dir, check=True
+            )
```

Comment:
I'm confused by this. Wouldn't the directory change be overridden by the `cwd` argument?